packages/algob/src/builtin-tasks/run.ts

```diff
@@ -82,31 +90,32 @@ export async function runMultipleScriptsOneByOne (
   allowWrite: boolean,
   algoOp: AlgoOperator
 ): Promise<void> {
-  for (const script of scriptNames) {
-    await runMultipleScripts(
+  const accounts = mkAccountIndex(runtimeEnv.network.config.accounts);
+  const asaDefs = loadASAFile(accounts);
+  for (const scriptsBatch of batchScriptNames(scriptNames)) {
+    await runScriptsBatch(
       runtimeEnv,
-      [script],
+      scriptsBatch,
       onSuccessFn,
       force,
       logDebugTag,
-      allowWrite,
-      algoOp
+      (cpData: CheckpointRepo) => mkDeployer(runtimeEnv, cpData, allowWrite, algoOp, asaDefs, accounts)
```

Comment:
1. The function doesn't change, so no point to redefine the function in the loop.
2. Can we get rid of this function? It seams that we don't need it - we can pass other arguments here (we can even use object to group them)

Reply:
My point is that we don't even need to use a function here.

Reply:
What else would you want to use to preserve loaded `runtimeEnv` and other arguments when `cpData` is reloaded inside of `runScript`?

Reply:
We can pass arguments (`env`, `allowWrite`) and construct the `deployer` in `runScriptsBatch`.

Reply:
Ok.
Added 4 parameters to the method to not create a function:
```
  allowWrite: boolean,
  algoOp: AlgoOperator,
  asaDefs: ASADefs,
  accounts: Accounts
```
Now the method takes 9 parameters of which 4 is exclusively used for `mkDeployer` function.

Reply:
Why you don't put this parameters for `mkDeployer` in an object as I suggested above? (in the first post)

Reply:
Then we'll need a type and I preferred to use functions so I used them. `runSortedScripts` doesn't need to know everything about internals of deployer so I prefer to remove what is not needed by it and make `runSortedScripts` not care instead of having huge objects that can be interacted from every place.
It's a viable strategy both ways but I preferred to know less where I don't need to know.

Reply:
And even when we'll add an object then either:
- Deployer.constructor will take it as a parameter (probably even add it to `types.ts` and make it public to everything)
- Destructure the object inside of the method but then everything is still visible to the method.
- Curried function (I used this way to have only one param instead of several)
- Pass everything -- Equivalent to object but without a type and without showing it to the whole app.
These ways are fine depending on the style that you prefer to use.

Reply:
With function you need both: a function and a types. 

Let's group arguments in an object: `DeployerConfig`, and store it `src/lib/deployer.ts`. This way the function will not need to know details, and will just pass the config, without having a responsibility to call external functions.

Reply:
You didn't understand what I meant by "know types".
There are two functions:
Outer one:`runMultipleScripts` and inner one: `runSortedScripts`.
The outer one does some IO and creates some variables.
The inner one does only use CPRepo and has to load it inside. It also needs Deployer to send it to the script method. That's it. It doesn't need to care about any other variables so I encapsulated it in a function.

The function type is simpler than the type which you propose because it takes only CPRepo and produces Deployer.
What you propose is to have a type with 4 or more fields that are still visible to the inner function.
It could be that this type signals that we have too much logic in Deployer already and we should limit it.

Now what you want is not only to show the variables to the inner function but to create a larger type in a completely different file to solve the issue with first file. If we will end up having 20 variables in DeployerImpl's constructor then it's probably an issue with DeployerImpl class and not with this function.

And the object will still need to be created so my function solution would still have worked in the exact same way.
Except instead of having a type with four or more fields I had a type with input and output. 
And my top-level `mkDeployer` function has more fields too.

Reply:
I alread restored the code to the method with many variables. And for now the DeployerImpl's constructor doesn't have many variables so the type there is not needed.

Reply:
>  I preferred to know less where I didn't need to know.

I agree

Reply:
Merge it and I will update it later, because it is taking too long.